R/part6AlignIndividuals.R

```diff
@@ -0,0 +1,186 @@
+      out2$N_valid_hhmembers = rowSums(out2[,grep(pattern = "validepoch", x = names(out2), value = FALSE)], na.rm = TRUE)
+      # Add indicator of valid data for a pair of family members per time point
+      varn_remember = c()
+      Nmembers = length(uMID)
+      for (pa1 in 1:Nmembers) {
+        for (pa2 in 1:Nmembers) {
+          if (pa1 != pa2 & pa2 > pa1) {
+            col1 = grep(pattern = paste0("validepoch.", uMID[pa1]), x = names(out2), value = FALSE)
+            col2 = grep(pattern = paste0("validepoch.", uMID[pa2]), x = names(out2), value = FALSE)
+            validpair = ifelse(test = rowSums(out2[, c(col1, col2)], na.rm = TRUE) == 2, yes = TRUE, no = FALSE)
+            out2$new = validpair
+            varname = paste0("validpair_",uMID[pa1],"_", uMID[pa2])
+            colnames(out2)[which(colnames(out2) == "new")] = varname
+            varn_remember = c(varn_remember, varname)
+          }
+        }
+      }
+      Npairs = length(varn_remember)
+      #======================================================================
+      # Visualise data
+      if (exists("currentColorPalette")) rm(currentColorPalette)
+      Ncols = (Npairs + Nmembers)
+      palette(rainbow(n = Ncols, alpha = 1)) 
+      currentColorPalette = palette()
+      colorsM = currentColorPalette[1:Nmembers]
+      colorsP = currentColorPalette[(Nmembers + 1):(Nmembers + Npairs)]
+      
+      par(mfrow = c(2,1))
+      plot(out2$time_POSIX, out2[, colnames[1]], type = "l", col = colorsM[1],
+           main = paste0("Houshold ", uHID[h]), xlab = "", ylab = "ACC", bty = "l")
+      for (jj in 2:length(colnames)) {
+        lines(out2$timenum, out2[, colnames[jj]], type = "l", col = colorsM[jj])
+      }
+      legend("topright", legend = colnames,
+             col = colorsM[1:length(colnames)], lty = 1, cex = 0.6, bg = "white", ncol = 3)
+      LTY = rep(1, Npairs)
+      
+      for (vn in 1:Npairs) {
+        if (vn == 1) {
+          plot(out2$time_POSIX, out2[,varn_remember[vn]] - 0.05 + (vn/50), type = "l",
+               col = colorsP[vn], ylim = c(0, 1.5),
+               xlab = "", ylab = "Valid pair", bty = "l",
+               axes = FALSE, lty = LTY[vn], lwd = 1.3)
+        } else {
+          lines(out2$time_POSIX, out2[,varn_remember[vn]] - 0.05 + (vn/50), type = "l",
+                col = colorsP[vn], lty = LTY[vn], lwd = 1.3)
+        }
+      }
+      axis(side = 2, at = c(0, 1), labels = c("FALSE", "TRUE"))
+      legend("topright", legend = gsub(pattern = "valid", replacement = "", x = varn_remember),
+             col = colorsP[1:Npairs], lty = LTY[1:Npairs], cex = 0.6, bg = "white", ncol = 3)
+      for (i in 1:ncol(out2)) {
+        NA_NaN = which(is.na(out2[, i]))
+        out2[NA_NaN, i] = ""
+      }
+      rm(currentColorPalette)
+      # Store data
+      write.csv(x = out2, file = paste0(path_results, "/alignedTimeseries/timeseries_HID_",uHID[h],".csv"), row.names = FALSE)
```

Comment:
Should we give the option to store this as `RData` to save space?

Reply:
sorry, forgot to respond to this... now looking into it.

Reply:
I am now changing this to store RData by default. The few users who may want to work with this data will just have to load the RData files. I think there is no major need to facilitate both csv and RData.